File: frontend/src/types/startup.ts
export interface StartupData {
  id: string;
  company_name: string;
  founded_year: number;
  industry: string;
  stage: string;
  description: string;
  team: TeamMember[];
  metrics: StartupMetrics;
  financial_data: FinancialData;
  market_data: MarketData;
  product_info: ProductInfo;
  traction: TractionData;
  funding_history: FundingRound[];
}

export interface TeamMember {
  name: string;
  role: string;
  bio: string;
  experience_years: number;
  previous_companies?: string[];
  education?: string[];
}

export interface StartupMetrics {
  revenue_arr?: number;
  monthly_growth_rate?: number;
  customer_count?: number;
  churn_rate?: number;
  ltv_cac_ratio?: number;
  gross_margin?: number;
  burn_rate?: number;
  runway_months?: number;
}

export interface FinancialData {
  current_valuation?: number;
  last_funding_amount?: number;
  total_funding_raised: number;
  burn_rate?: number;
  runway_months?: number;
  revenue_streams: string[];
  unit_economics: {
    cac?: number;
    ltv?: number;
    payback_period_months?: number;
  };
}

export interface MarketData {
  total_addressable_market: number;
  serviceable_addressable_market: number;
  target_market_size: number;
  market_growth_rate: number;
  competitors: Competitor[];
  market_position: string;
}

export interface Competitor {
  name: string;
  description: string;
  funding_raised?: number;
  employee_count?: number;
  market_share?: number;
}

export interface ProductInfo {
  product_name: string;
  product_type: string;
  key_features: string[];
  technology_stack?: string[];
  intellectual_property: string[];
  development_stage: string;
}

export interface TractionData {
  user_growth_metrics: {
    monthly_active_users?: number;
    user_growth_rate?: number;
    user_retention_rate?: number;
  };
  business_metrics: {
    revenue_growth?: number;
    customer_acquisition_rate?: number;
    market_penetration?: number;
  };
  partnerships: Partnership[];
  awards_recognition: string[];
}

export interface Partnership {
  partner_name: string;
  partnership_type: string;
  description: string;
  value?: string;
}

export interface FundingRound {
  round_type: string;
  amount: number;
  date: string;
  lead_investor?: string;
  participating_investors: string[];
  valuation?: number;
  use_of_funds: string[];
}

export interface BenchmarkData {
  peer_group: PeerCompany[];
  industry_averages: IndustryMetrics;
  percentile_rankings: PercentileRankings;
  growth_comparisons: GrowthComparison[];
}

export interface PeerCompany {
  name: string;
  stage: string;
  industry: string;
  metrics: StartupMetrics;
  last_funding_date: string;
  total_funding: number;
}

export interface IndustryMetrics {
  median_arr_growth: number;
  median_churn_rate: number;
  median_cac_payback: number;
  median_gross_margin: number;
  typical_funding_amounts: Record<string, number>;
}

export interface PercentileRankings {
  revenue_growth: number;
  customer_growth: number;
  funding_efficiency: number;
  team_experience: number;
  market_position: number;
}

export interface GrowthComparison {
  metric: string;
  company_value: number;
  peer_median: number;
  peer_75th_percentile: number;
  peer_90th_percentile: number;
  ranking: 'top_10' | 'top_25' | 'median' | 'below_median';
}

export interface RiskAssessment {
  overall_risk_score: number;
  risk_factors: RiskFactor[];
  red_flags: RedFlag[];
  yellow_flags: YellowFlag[];
  risk_mitigation_suggestions: string[];
}

export interface RiskFactor {
  category: 'market' | 'team' | 'financial' | 'product' | 'competitive' | 'regulatory';
  risk_type: string;
  severity: 'low' | 'medium' | 'high' | 'critical';
  description: string;
  evidence: string[];
  impact_description: string;
  likelihood: number;
  potential_impact: number;
}

export interface RedFlag {
  type: string;
  description: string;
  evidence: string[];
  severity_score: number;
  recommendation: string;
}

export interface YellowFlag {
  type: string;
  description: string;
  evidence: string[];
  monitoring_suggestion: string;
}

export interface InvestorMemo {
  id: string;
  company_name: string;
  generated_at: string;
  executive_summary: ExecutiveSummary;
  investment_thesis: InvestmentThesis;
  market_analysis: MarketAnalysis;
  team_assessment: TeamAssessment;
  financial_analysis: FinancialAnalysis;
  risk_analysis: RiskAnalysisSection;
  recommendation: InvestmentRecommendation;
  appendix: MemoAppendix;
}

export interface ExecutiveSummary {
  company_overview: string;
  key_highlights: string[];
  investment_highlights: string[];
  concerns: string[];
  recommendation_summary: string;
}

export interface InvestmentThesis {
  value_proposition: string;
  market_opportunity: string;
  competitive_advantage: string[];
  growth_potential: string;
  exit_strategy: string;
}

export interface MarketAnalysis {
  market_size_assessment: string;
  growth_projections: string;
  competitive_landscape: string;
  market_positioning: string;
  customer_validation: string;
}

export interface TeamAssessment {
  leadership_strength: string;
  team_completeness: string;
  execution_capability: string;
  domain_expertise: string;
  concerns: string[];
}

export interface FinancialAnalysis {
  revenue_model_assessment: string;
  unit_economics_analysis: string;
  growth_trajectory: string;
  funding_requirements: string;
  burn_rate_analysis: string;
  runway_assessment: string;
}

export interface RiskAnalysisSection {
  key_risks: string[];
  risk_mitigation: string;
  red_flags_summary: string;
  monitoring_recommendations: string[];
}

export interface InvestmentRecommendation {
  recommendation: 'strong_buy' | 'buy' | 'hold' | 'pass' | 'strong_pass';
  confidence_level: number;
  reasoning: string[];
  suggested_valuation_range: {
    low: number;
    high: number;
  };
  investment_amount_suggestion: number;
  terms_suggestions: string[];
  next_steps: string[];
}

export interface MemoAppendix {
  raw_data: any;
  methodology: string;
  assumptions: string[];
  data_sources: string[];
  confidence_scores: {
    financial_analysis: number;
    market_analysis: number;
    team_assessment: number;
    risk_analysis: number;
  };
}

File: backend/app/models/startup.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
from datetime import datetime
from enum import Enum

# Enums
class JobStatus(str, Enum):
    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"

class JobStage(str, Enum):
    UPLOAD = "upload"
    OCR = "ocr"
    PARSING = "parsing"
    BENCHMARK = "benchmark"
    RISKS = "risks"
    MEMO = "memo"

class RiskSeverity(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    CRITICAL = "critical"

class RiskCategory(str, Enum):
    MARKET = "market"
    TEAM = "team"
    FINANCIAL = "financial"
    PRODUCT = "product"
    COMPETITIVE = "competitive"
    REGULATORY = "regulatory"

class InvestmentRecommendation(str, Enum):
    STRONG_BUY = "strong_buy"
    BUY = "buy"
    HOLD = "hold"
    PASS = "pass"
    STRONG_PASS = "strong_pass"

# Base Models
class TimestampMixin(BaseModel):
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

class TeamMember(BaseModel):
    name: str
    role: str
    bio: Optional[str] = None
    experience_years: Optional[int] = None
    previous_companies: Optional[List[str]] = []
    education: Optional[List[str]] = []

class StartupMetrics(BaseModel):
    revenue_arr: Optional[float] = None
    monthly_growth_rate: Optional[float] = None
    customer_count: Optional[int] = None
    churn_rate: Optional[float] = None
    ltv_cac_ratio: Optional[float] = None
    gross_margin: Optional[float] = None
    burn_rate: Optional[float] = None
    runway_months: Optional[int] = None

class UnitEconomics(BaseModel):
    cac: Optional[float] = None
    ltv: Optional[float] = None
    payback_period_months: Optional[int] = None

class FinancialData(BaseModel):
    current_valuation: Optional[float] = None
    last_funding_amount: Optional[float] = None
    total_funding_raised: Optional[float] = None
    burn_rate: Optional[float] = None
    runway_months: Optional[int] = None
    revenue_streams: List[str] = []
    unit_economics: UnitEconomics = UnitEconomics()

class Competitor(BaseModel):
    name: str
    description: str
    funding_raised: Optional[float] = None
    employee_count: Optional[int] = None
    market_share: Optional[float] = None

class MarketData(BaseModel):
    total_addressable_market: Optional[float] = None
    serviceable_addressable_market: Optional[float] = None
    target_market_size: Optional[float] = None
    market_growth_rate: Optional[float] = None
    competitors: List[Competitor] = []
    market_position: Optional[str] = None

class ProductInfo(BaseModel):
    product_name: Optional[str] = None
    product_type: Optional[str] = None
    key_features: List[str] = []
    technology_stack: Optional[List[str]] = []
    intellectual_property: List[str] = []
    development_stage: Optional[str] = None

class Partnership(BaseModel):
    partner_name: str
    partnership_type: str
    description: str
    value: Optional[str] = None

class UserGrowthMetrics(BaseModel):
    monthly_active_users: Optional[int] = None
    user_growth_rate: Optional[float] = None
    user_retention_rate: Optional[float] = None

class BusinessMetrics(BaseModel):
    revenue_growth: Optional[float] = None
    customer_acquisition_rate: Optional[float] = None
    market_penetration: Optional[float] = None

class TractionData(BaseModel):
    user_growth_metrics: UserGrowthMetrics = UserGrowthMetrics()
    business_metrics: BusinessMetrics = BusinessMetrics()
    partnerships: List[Partnership] = []
    awards_recognition: List[str] = []

class FundingRound(BaseModel):
    round_type: str
    amount: float
    date: str
    lead_investor: Optional[str] = None
    participating_investors: List[str] = []
    valuation: Optional[float] = None
    use_of_funds: List[str] = []

class StartupData(BaseModel):
    id: str
    company_name: str
    founded_year: int
    industry: str
    stage: str
    description: str
    team: List[TeamMember] = []
    metrics: StartupMetrics
    financial_data: FinancialData
    market_data: MarketData
    product_info: ProductInfo
    traction: TractionData
    funding_history: List[FundingRound] = []

class RiskFactor(BaseModel):
    category: RiskCategory
    risk_type: str
    severity: RiskSeverity
    description: str
    evidence: List[str] = []
    impact_description: str
    likelihood: float = Field(ge=0, le=1)
    potential_impact: int = Field(ge=1, le=10)

class RedFlag(BaseModel):
    type: str
    description: str
    evidence: List[str] = []
    severity_score: int = Field(ge=1, le=10)
    recommendation: str

class YellowFlag(BaseModel):
    type: str
    description: str
    evidence: List[str] = []
    monitoring_suggestion: str

class RiskAssessment(BaseModel):
    overall_risk_score: int = Field(ge=0, le=100)
    risk_factors: List[RiskFactor] = []
    red_flags: List[RedFlag] = []
    yellow_flags: List[YellowFlag] = []
    risk_mitigation_suggestions: List[str] = []

class Job(BaseModel):
    id: str
    status: JobStatus
    stage: JobStage
    progress: Optional[int] = Field(default=None, ge=0, le=100)
    message: Optional[str] = None
    file_id: Optional[str] = None
    result: Optional[Dict[str, Any]] = None
    error: Optional[str] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)